src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -307,55 +435,53 @@ static void CheckForExperimentStatus(ITextView textView, in CompletionOptions op
                 itemsBuilder.Add(item);
             }
 
-            AddPropertiesToSession(session, completionList, triggerLocation);
-
             var filters = filterSet.GetFilterStatesInSet();
-            var items = itemsBuilder.ToImmutableAndFree();
+            var items = itemsBuilder.ToImmutable();
 
             if (completionList.SuggestionModeItem is null)
-                return new(items, suggestionItemOptions: null, selectionHint: AsyncCompletionData.InitialSelectionHint.RegularSelection, filters);
+                return (new(items, suggestionItemOptions: null, selectionHint: AsyncCompletionData.InitialSelectionHint.RegularSelection, filters), completionList);
 
             var suggestionItemOptions = new AsyncCompletionData.SuggestionItemOptions(
                 completionList.SuggestionModeItem.DisplayText,
                 completionList.SuggestionModeItem.Properties.TryGetValue(Description, out var description) ? description : string.Empty);
 
-            return new(items, suggestionItemOptions, selectionHint: AsyncCompletionData.InitialSelectionHint.SoftSelection, filters);
+            return (new(items, suggestionItemOptions, selectionHint: AsyncCompletionData.InitialSelectionHint.SoftSelection, filters), completionList);
+        }
 
-            static void AddPropertiesToSession(IAsyncCompletionSession session, CompletionList completionList, SnapshotPoint triggerLocation)
+        private static void AddPropertiesToSession(IAsyncCompletionSession session, CompletionList completionList, SnapshotPoint triggerLocation)
```

Comment:
💡 Can you add a comment on this explaining in fair detail what it's purpose is?

Reply:
➡️ Looks like this didn't change as part of this PR (just indentation).

Reply:
I will leave it as is since I'm hoping to get rid the property bag usage in a follow up.